src/freenet/clients/http/FProxyToadlet.java

```diff
@@ -984,6 +984,13 @@ public void run() {
 		}
 	}
 
+	private String getSchemeHostAndPort(ToadletContext ctx) {
```

Comment:
This feels unfinished because it doesn’t do what the name says it’ll do and it’s also not used like it does what the name says it does. Please finish.

Reply:
yes, it is unfinished. The required changes for Forwarded grew bigger.

Reply:
Now it’s in, but I cannot test it, because I don’t have a proxy to check.

Reply:
You can use `curl` or `wget` to create a request with the respective header fields set, a proxy is not strictly necessary. :)

Reply:
Heh, while it is totally safer to only allow known values for those headers it will break my setup (because Fred doesn’t know anything about being reverse-proxied). That can probably be added in a future PR with a config option.

However, I still would like to see this extracted into an extra component and properly unit-tested.

Reply:
For your setup you’ll need to add the proxy-url to the bindTo addresses. Then fred will allow it. This is a stop-gap measure, but it should work.

Reply:
That’s… not how reverse proxies work, and I won’t bind Fred to an external IP address. Anyway, later PR. :)

Reply:
sounds good — I extracted the code now, added some tests and found that default values for configured IP and Port were missing (that case should not be possible, but "should not").